### PR TITLE
Improve function_call and maybe_consume_extra parsing speed

### PR DESCRIPTION
**Pull request by SanderJSA**

### CODE REVIEW

File: src/parser/constructs.rs [hunk 1]
```diff
@@ -163,43 +151,26 @@ impl Construct {
...
         }
 
-        // Get 0 or more arguments with a comma to the function call
-        let (input, mut arg_vec) = many0(arg_and_comma)(input)?;
-
-        // Parse the last argument, which does not have a comma. There needs to be
-        // at least one argument, which can be this one
-        let (input, last_arg) = Construct::arg(input)?;
+        // Get first arg
+        let (input, arg) = Construct::arg(input)?;
```
Comment (CohenArthur) [inline src/parser/constructs.rs:164]: Could we call this `first_arg` instead?

File: src/parser/constructs.rs [hunk 2]
```diff
@@ -273,12 +244,18 @@ impl Construct {
     /// `<arg_list> := [(<constant> | <variable> | <instruction>)*]`
     /// `<identifier> ( <arg_list> )`
     pub(crate) fn function_call(input: &str) -> ParseResult<&str, FunctionCall> {
-        let call = alt((
-            Construct::function_call_no_args,
-            Construct::function_call_args,
-        ))(input)?;
+        let (input, fn_id) = Token::identifier(input)?;
+        let (input, _) = Token::left_parenthesis(input)?;
```
Comment (CohenArthur) [inline src/parser/constructs.rs:248]: Hey, there's a bug here :D (Which was present in the original impl, no worries)

Could you add a test with the following function call and declaration (in another PR if possible, or I can also fix it but if you don't want to do it I'll do it :) )
```rust
func fn_name () {}
fn_name ()
// Note the extra whitespaces before the parentheses
```

**Review by CohenArthur — APPROVED** [review]
:rocket: :rocket: :rocket: :rocket: :rocket: :rocket:

**Comment by CohenArthur** [comment]
Also @SanderJSA, don't worry about the coverage failing. The difference is very small so that's completely fine. Most refactors lower the coverage for some reason

File: src/parser/constructs.rs [hunk 3]
```diff
@@ -163,43 +151,26 @@ impl Construct {
     /// Parse a list of arguments separated by comma
     fn args_list(input: &str) -> ParseResult<&str, Vec<Box<dyn Instruction>>> {
         /// Parse an argument and the comma that follows it
```
Comment (Skallwar) [inline src/parser/constructs.rs:153]:
```suggestion
        /// Parse a comma and a following argument
```

**Review by Skallwar — APPROVED** [review]
Maybe rename the pull request to reflect what in the parser get a speed improvement. Great work :rocket:

**Review by CohenArthur — APPROVED** [review]
:rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket: :rocket:

**Comment by SanderJSA** [comment]
Thanks !
Changed PR name to better reflect modifications made.

**Pull request merged**